app/forms/questionnaire_form.py

```diff
@@ -43,9 +43,13 @@ def validate(self):
 
         for question in self.schema.get_questions_for_block(self.block_json):
             if question['type'] == 'DateRange':
-                valid_date_range_form = self.validate_date_range_question(question)
+                valid_date_range = self.validate_date_range_question(question)
+                if valid_date_range_form:
```

Comment:
@dwyeradam Wont this always be true? Same for below also

Reply:
Not if the page has multiple DateRange or Calculation type questions. Very edge case but while I was here. Similar to you changing the names to stop them overriding each other.